receiver/mongodbatlasreceiver/receiver_test.go

```diff
@@ -71,3 +71,64 @@ func TestTimeConstraints(t *testing.T) {
 		t.Run(testCase.name, testCase.run)
 	}
 }
+
+func TestShouldProcessCluster(t *testing.T) {
+	tests := []struct {
+		name        string
+		projectCfg  *ProjectConfig
+		clusterName string
+		want        bool
+	}{
+		{
+			name: "included cluster",
+			projectCfg: &ProjectConfig{
+				IncludeClusters: []string{"Cluster1"},
+				includesByClusterName: map[string]struct{}{
+					"Cluster1": {},
+				},
+			},
+			clusterName: "Cluster1",
+			want:        true,
+		},
+		{
+			name: "excluded cluster",
+			projectCfg: &ProjectConfig{
+				ExcludeClusters: []string{"Cluster2"},
+				excludesByClusterName: map[string]struct{}{
+					"Cluster2": {},
+				},
+			},
+			clusterName: "Cluster2",
+			want:        false,
+		},
+		{
+			name: "cluster not in include or exclude list",
```

Comment:
Is this a valid case? Doesn't the config validation require that at max one of include/exclude lists be used?

Reply:
I see, I updated the valid cases. I revised the test cases based on the config `Validate` function which when called asserts that there won't be an included and excluded cluster at the same time with:
```go
		if len(project.ExcludeClusters) != 0 && len(project.IncludeClusters) != 0 {
			errs = multierr.Append(errs, errClusterConfig)
		}
```

and `populateIncludesAndExcludes` function is called so I only need to put in the `IncludeClusters` or `ExcludeClusters` values:
```go
func (pc *ProjectConfig) populateIncludesAndExcludes() {
	pc.includesByClusterName = map[string]struct{}{}
	for _, inclusion := range pc.IncludeClusters {
		pc.includesByClusterName[inclusion] = struct{}{}
	}

	pc.excludesByClusterName = map[string]struct{}{}
	for _, exclusion := range pc.ExcludeClusters {
		pc.excludesByClusterName[exclusion] = struct{}{}
	}
}
```